homeassistant/components/imap/coordinator.py

```diff
@@ -92,13 +92,15 @@ async def idle_loop(self) -> None:
                     self.config_entry.data[CONF_SERVER],
                 )
                 self.imap_client = None
+                if self.idle_loop_task:
```

Comment:
We're inside the task so we shouldn't cancel ourselves. Just let the loop exit by breaking and scheduling the refresh request instead of awaiting it.

Reply:
So it should be like this?
```python
        break
asyncio.create_task(self.async_request_refresh())
```

Reply:
I think so, yes.

Reply:
But use `hass.async_create_task` as it's not a long running task.